### PR TITLE
Av2 846/notification preference desktop

### PR DESCRIPTION
**Pull request by ronakamani**
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-846

**Additional Comments:**
`Add here`

**Checklist:**

- [ x ] Unit test completed?: (Y/N)
- [ x ] Docs folder up to date?: (Y/N)
- [ x ] Add if anything missed: (Y/N)

All Checklist are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

### CODE REVIEW

File: src/app/preference/preference-modal/preference-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<ion-content color="light" class="pop-up-content test" text-center>
```
Comment (trtshen) [inline src/app/preference/preference-modal/preference-modal.component.html]: @ronakamani is the "test" necessary?

File: src/app/preference/preference-modal/preference-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  <ng-container *ngIf="!showUpdateModal; else showUpdateModalTemplate">
+    <ion-toolbar class="pn-title-toolbar">
```
Comment (trtshen) [inline src/app/preference/preference-modal/preference-modal.component.html:3]: @ronakamani i dont recommend to copy over the entire [preference.component.ts](https://github.com/intersective/app-v2/blob/AV2-846/Notification-preference-desktop/src/app/preference/preference.component.html) into a different file. it reduces code quality and maintainability.

how about try learn to use [Template syntax](https://angular.io/guide/architecture-components#template-syntax) to insert the instance of component view. 

In your case, 
eg. 
```html
<app-preference-component></app-preference-component>
```

**Review by trtshen — CHANGES_REQUESTED** [review]

File: src/app/preference/preference-modal/preference-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      name: '',
+      description: '',
+      options: '',
+      remarks: '',
+    };
+  }
+  close () {
+    this.modalController.dismiss();
+  }
+  PreferenceUpdateModal(event, key) {
```
Comment (trtshen) [inline src/app/preference/preference-modal/preference-modal.component.ts]: the naming convention should follow [camelCase](https://www.w3schools.com/js/js_conventions.asp#:~:text=camelCase%3A,with%20many%20JavaScript%20library%20names.)

File: src/app/preference/preference-modal/preference-modal.component.html [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    <ng-template #noPreferenceAvailable>
+      <div class="ion-text-center ion-margin-bottom">
+        <p>No preferences setting available.</p>
+      </div>
+    </ng-template>
+  </ng-container>
+ 
+  
+  <ng-template #showUpdateModalTemplate>
+    <ion-header mode="ios">
```
Comment (trtshen) [inline src/app/preference/preference-modal/preference-modal.component.html]: it's should be the same for this  `#showUpdateModalTemplate` part.
this part using the entire same codebase as in [preference-update.component.html](https://github.com/intersective/app-v2/blob/AV2-846/Notification-preference-desktop/src/app/preference/preference-update/preference-update.component.html)

try practise [template syntax](https://angular.io/guide/architecture-components#template-syntax)

p/s: anything that's repeated, should be made as shareable "component". Next time, we dont need to edit multiple time/components for one similar change request.

File: src/app/shared/notification/notification.service.ts [hunk 5]
```diff
@@ -194,4 +195,18 @@ export class NotificationService {
     });
     return loading.present();
   }
+
+  async PreferenceDesktopModal(message: string) {
```
Comment (trtshen) [inline src/app/shared/notification/notification.service.ts]: naming convention follow [camelCase](https://www.w3schools.com/js/js_conventions.asp#:~:text=camelCase%3A,with%20many%20JavaScript%20library%20names.)

**Review by trtshen — CHANGES_REQUESTED** [review]

**Comment by ronakamani** [comment]
The UI had problem to be implemented. We decided to stop implementing in this way.

**Pull request closed without merging**